### ISSUE TITLE
Starting mock SQS (http port 4576)...Error: Invalid or corrupt jarfile DynamoDBLocal.jar

### ISSUE THREAD

**Issue by tho1** [localstack/localstack#1575]
Hi,
I am trying to run localstack locally.  I issued the command DEBUG=1 localstack start --host and it keeps on failing on the dynamo module.  It mentioned issue with the jar DynamoDBLocal.jar.  Where can I find a good one?

Here is the log:

start with the command _"DEBUG=1 localstack start --host"_

Starting local dev environment. CTRL-C to quit.
No module named services
Starting mock SES (http port 4579)...
2019-09-19T08:02:01:INFO:localstack.multiserver: Starting multi API server process on port 51492
Starting mock CloudWatch Logs (http port 4586)...
2019-09-19 08:02:02,034:API:  * Running on http://0.0.0.0:4579/ (Press CTRL+C to quit)
2019-09-19 08:02:02,034:API:  * Running on http://0.0.0.0:4579/ (Press CTRL+C to quit)
2019-09-19 08:02:02,034:API:  * Running on http://0.0.0.0:4579/ (Press CTRL+C to quit)
Starting mock Redshift (http port 4577)...
2019-09-19 08:02:02,043:API:  * Running on http://0.0.0.0:4586/ (Press CTRL+C to quit)
2019-09-19 08:02:02,043:API:  * Running on http://0.0.0.0:4586/ (Press CTRL+C to quit)
2019-09-19 08:02:02,043:API:  * Running on http://0.0.0.0:4586/ (Press CTRL+C to quit)
Starting mock CloudFormation (http port 4581)...
2019-09-19 08:02:02,047:API:  * Running on http://0.0.0.0:4577/ (Press CTRL+C to quit)
2019-09-19 08:02:02,047:API:  * Running on http://0.0.0.0:4577/ (Press CTRL+C to quit)
2019-09-19 08:02:02,047:API:  * Running on http://0.0.0.0:4577/ (Press CTRL+C to quit)
Starting mock EC2 (http port 4597)...
2019-09-19 08:02:02,053:API:  * Running on http://0.0.0.0:4559/ (Press CTRL+C to quit)
2019-09-19 08:02:02,053:API:  * Running on http://0.0.0.0:4559/ (Press CTRL+C to quit)
2019-09-19 08:02:02,053:API:  * Running on http://0.0.0.0:4559/ (Press CTRL+C to quit)
Starting mock Firehose service (http port 4573)...
Starting mock Route53 (http port 4580)...
Starting mock ES service (http port 4578)...
Starting mock IAM (http port 4593)...
Starting mock DynamoDB Streams service (http port 4570)...
Starting mock S3 (http port 4572)...
Starting mock CloudWatch (http port 4582)...
Starting mock DynamoDB (http port 4569)...
Starting mock CloudWatch Events (http port 4587)...
Starting mock Kinesis (http port 4568)...
Starting mock SQS (http port 4576)...Error: Invalid or corrupt jarfile DynamoDBLocal.jar

Starting mock SSM (http port 4583)...
Starting mock SNS (http port 4575)...
Error: Unable to access jarfile /usr/lib/python2.7/site-packages/localstack/infra/elasticmq/elasticmq-server.jar
Starting mock API Gateway (http port 4567)...
Starting mock Secrets Manager (http port 4584)...
Starting local Elasticsearch (http port 4571)...
Starting mock STS (http port 4592)...
Starting mock StepFunctions (http port 4585)...
Starting mock Lambda service (http port 4574)...
su: unknown user localstack
Error: Unable to access jarfile StepFunctionsLocal.jar
Listening at http://:::4565
2019-09-19 08:02:12,561:API: 127.0.0.1 - - [19/Sep/2019 08:02:12] "GET / HTTP/1.1" 200 -
2019-09-19T08:02:23:WARNING:infra.pyc: Service "dynamodb" not yet available, retrying...
2019-09-19 08:02:26,590:API: 127.0.0.1 - - [19/Sep/2019 08:02:26] "GET / HTTP/1.1" 200 -
2019-09-19T08:02:37:WARNING:infra.pyc: Service "dynamodb" not yet available, retrying...
2019-09-19 08:02:40,620:API: 127.0.0.1 - - [19/Sep/2019 08:02:40] "GET / HTTP/1.1" 200 -
2019-09-19T08:02:51:WARNING:infra.pyc: Service "dynamodb" not yet available, retrying...
2019-09-19 08:02:54,653:API: 127.0.0.1 - - [19/Sep/2019 08:02:54] "GET / HTTP/1.1" 200 -
2019-09-19T08:03:05:WARNING:infra.pyc: Service "dynamodb" not yet available, retrying...

**Comment by AlexisWilke** [comment]
I'm also having this port 4565 problem. Also when I look at `netstat -a64n | grep LISTEN` nothing is listening on ports 4564 or 4565. DynamoDB doesn't start at all. See #1689

**Comment by whummer** [comment]
Thanks for reporting @tho1 @AlexisWilke . Please note that the recommended/supported way of starting LocalStack these days is in Docker. If you need to run on the host, you can try to delete the `localstack/infra/elasticmq` folder of your LocalStack installation, which should trigger an automatic re-download on the next LocalStack restart.

Closing this issue - please report here if the problem persists. Thanks

**Comment by tho1** [comment]
Thanks whummer. I was running the localStack in Linux itself, not insider Docker. I will try the recommending method of removing elasticmq and see.

**Comment by tho1** [comment]
Just want to add a note to whoever run into this problem.  I found the dynamodbLocal.jar in the directory /usr/lib/python2.7/site-packages/localstack/infra/dynamodb. When I expand it, it only contains the file log4j2.xml.  So I went to https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.DownloadingAndRunning.htm, and download the tar.gz file in Step 1.  I picked up US Oregon, and inside the tar.gz file contains the DynamoLocal.jar.  I replaced the existing jar file, and then it works. 

Keep in mind this is an older version of localstack.  0.10.2
